File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {

    public double productionRatePerHour(int speed) {
        double baseProduction = speed * 221.0;
        double successRate;
        
        if (speed >= 1 && speed <= 4) {
            successRate = 1.0;
        } else if (speed >= 5 && speed <= 8) {
            successRate = 0.9;
        } else if (speed == 9) {
            successRate = 0.8;
        } else if (speed == 10) {
            successRate = 0.77;
        } else {
            // speed 0 or invalid input
            successRate = 0.0;
        }
        
        return baseProduction * successRate;
    }

    public int workingItemsPerMinute(int speed) {
        return (int) (productionRatePerHour(speed) / 60);
    }
}

File: solutions/r/cars-assemble/1/cars-assemble.R
success_rate <- function(speed) {
  if (speed == 0) {
    0
  } else if (speed <= 4) {
    1.0
  } else if (speed <= 8) {
    0.9
  } else if (speed == 9) {
    0.8
  } else {  # speed == 10
    0.77
  }
}

production_rate_per_hour <- function(speed) {
  # Base production: speed * 221 cars per hour
  # Multiply by success rate to get working cars
  speed * 221 * success_rate(speed)
}

working_items_per_minute <- function(speed) {
  # Get production rate per hour, divide by 60 to get per minute
  # Use floor (or integer division) to get whole cars
  floor(production_rate_per_hour(speed) / 60)
}